pkg/ddl/multi_schema_change_test.go

```diff
@@ -847,6 +849,44 @@ func TestMultiSchemaChangeMDLView(t *testing.T) {
 	tk.MustExec("commit;")
 }
 
+func TestMultiSchemaChangeWithoutMDL(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	defer func() {
+		tk.MustExec("set global tidb_enable_metadata_lock = default;")
+	}()
+
+	testcases := []struct {
+		name string
+		sql  string
+	}{
+		{"drop column", "alter table t drop column col2, drop column col3"},
+		{"drop index", "alter table t drop index idx2, drop index idx3"},
+		{"modify column", "alter table t modify column col2 bigint, modify column col3 bigint"},
+		{"modify column with reorg", "alter table t modify column col2 varchar(4) charset utf8mb4;"},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				require.Eventually(t, func() bool {
+					res := tk.MustQuery("admin show ddl jobs 1").Rows()
+					return res[0][11] == "synced"
+				}, 3*time.Second, 500*time.Millisecond)
+			}()
+			tk.MustExec("set global tidb_enable_metadata_lock = on;")
+			tk.MustExec("drop table if exists t;")
+			tk.MustExec("create table t(col1 int, col2 int, col3 int, index idx2(col2), index idx3(col2));")
+			tk.MustExec("set global tidb_enable_metadata_lock = off;")
+			tk.MustExec(tc.sql)
+			wg.Wait()
```

Comment:
If `MustExec` returns, the job state must be synced. I think we can remove the goroutine?